### ISSUE TITLE
Save in a cookie the language chosen by the reader

### ISSUE THREAD

**Issue by JaimeObregon** [cakephp/docs#5375]
First things first: **a big thank you!** For your efforts writing and maintaining the docs.

A suggestion: **Save in a cookie the language** chosen by the reader.

Why?

As a non native English speaker, my browser is set up in Spanish. Thus, googling for i.e `cakephp 2 html helper api`drives me to [/2.0/es/core-libraries/helpers/html.html](https://book.cakephp.org/2.0/es/core-libraries/helpers/html.html) (notice the `es` part).

But that page isn't translated yet. And even if it was, I'd rather prefer to read it in English. So I have to manually change the language every time. After a few dozens queries a day, this impacts the usability of the docs.

<img width="200" alt="captura de pantalla 2017-11-15 a las 12 56 55" src="https://user-images.githubusercontent.com/417226/32835088-116f6f02-ca05-11e7-999d-4c03c23295af.png">

Going further, I'd suggest to reply HTTP status code `404` in pages not translated yet, so Google doesn't crawl and index them. The content displayed might be the same –a placeholder suggesting to contribute a translation– but search engines and user agents would treat it as a "not found", which is what it actually is.

<img width="891" alt="captura de pantalla 2017-11-15 a las 12 53 17" src="https://user-images.githubusercontent.com/417226/32834748-fbcef0a6-ca03-11e7-8fa9-2f786b2e46d4.png">

**Comment by lucialovegood** [comment]
For me its also really annoying. The spanish documentation it is not complete and these make me change the url from es to en everytime I look for an answer. I would appreciate if you can improve it by setting a language cookie.

**Comment by markstory** [comment]
How would the language cookie work? The documentation is currently a static HTML site, so any logic we want to add will need to be done in nginx rules.  I've personally not done this before, but the nginx configuration for the docs is in this repository.

**Comment by JaimeObregon** [comment]
As a user, I'd expect to select my language preference once and then have the site stick to that preference from then on.

A possible solution could be to set a cookie by means of JavaScript, capturing the event when the reader chooses a language, and then have another piece of JavaScript read that cookie on every page load and redirect to the page in the preferred language. Possibly not ideal, as it would redirect to pages which might not exist translated.

I am not familiar with nginx rules, but I can set up and submit a pull-request of a prototype of the JavaScript alternative if it helps.

**Comment by markstory** [comment]
Do you know of a documentation site that does what you'd like to see well? I'm open to a Javascript based solution if the solution doesn't frustrate more people than it helps.

**Comment by JaimeObregon** [comment]
I have done a little research:

* `developer.mozilla.org` and `php.net` work both like `book.cakephp.org`: language selector, no language preference saved.
* `developers.google.com` and `support.google.com`: no language selector, no language preference saved.

The first three sites serve each translation in an independent URL. The two run by Google serve all the translations from the single URL, but the language can be changed via an optional query string: `?hl=es`.

Moreover, they all –and CakePHP docs too– actually complain with [Google recommendation on internationalization of content](https://support.google.com/webmasters/answer/189077).

So no portal seems to persist the language selection in the way I originally suggested. It looks like the ideal solution to this issue is to change my browser settings to prefer content in English, although this will affect all my browsing and not just CakePHP docs.

Seems a fair tradeoff to me. I'm closing this. Thank you!

**Comment by JaimeObregon** [comment]
For any future reader facing the same issue: it actually wasn't as simple as to set English as the preferred language in Chrome settings. Google still returned the docs in Spanish, even after a browser restart.

Instead I tried to change Google Search settings, but oddly enough it does not allow to deselect your first language, be that Spanish, English or other.

The only solution I have found, besides switching to DuckDuckGo, has been to change the language of *all* Google products to English. Only then I was finally able to force search results in English:

![captura de pantalla 2017-11-23 a las 22 51 19](https://user-images.githubusercontent.com/417226/33189204-e7994d04-d0a0-11e7-8924-86240b6a0274.png)

Far from ideal, since this affects *all* of my browsing, while I only want to read in English the CakePHP docs.
